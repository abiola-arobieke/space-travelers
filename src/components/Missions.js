import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAllMissions } from '../redux/missions/missionSlice';
import MissionsList from './MissionsList';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchAllMissions('https://api.spacexdata.com/v3/missions'));
  }, [dispatch]);

  return (
    <>
      <MissionsList missions={missions} />
    </>
  );
};

export default Missions;
