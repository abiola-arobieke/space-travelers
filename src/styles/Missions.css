.missions-table {
  border-collapse: collapse;
  margin-top: 1.5rem;
}

.mission-th,
.mission-td {
  border: 1px solid #ccc;
  padding: 8px;
}

.row-cell:nth-child(odd) {
  background-color: #cccccc30;
}

.member-button {
  font-size: 0.8rem;
  padding: 0.1rem 0.2rem;
}

.join-button {
  display: block;
  margin: auto;
}

.column-1 {
  width: 8rem;
}

.column-2 {
  width: 55rem;
}

.column-4 {
  width: 9.5rem;
}

.mission-description {
  font-size: 0.85rem;
  font-weight: normal;
}

.mission-name {
  vertical-align: top;
}
