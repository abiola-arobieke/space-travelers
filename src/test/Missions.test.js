import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Missions from '../components/Missions';

// Mock Redux store
const mockStore = configureStore([]);
const store = mockStore({ missions: [] });

describe('Missions', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
