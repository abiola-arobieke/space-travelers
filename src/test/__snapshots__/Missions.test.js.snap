// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly 1`] = `
<table
  className="missions-table"
>
  <thead>
    <tr>
      <th
        className="mission-th column-1"
      >
        Mission
      </th>
      <th
        className="mission-th column-2"
      >
        Description
      </th>
      <th
        className="mission-th column-3"
      >
        Status
      </th>
      <th
        className="mission-th column-4"
      >
         
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
